packages/server/modules/workspaces/repositories/sso.ts

```diff
@@ -1,53 +1,54 @@
+import { oidcProvider } from '@/modules/workspaces/domain/sso/models'
 import {
-  oidcProvider,
+  AssociateSsoProviderWithWorkspace,
   GetOIDCProviderData,
+  GetWorkspaceSsoProvider,
   StoreOIDCProviderValidationRequest,
   StoreProviderRecord,
+  UpsertUserSsoSession
+} from '@/modules/workspaces/domain/sso/operations'
+import {
   ProviderRecord,
-  AssociateSsoProviderWithWorkspace,
-  StoreUserSsoSession,
-  UserSsoSession,
-  GetWorkspaceSsoProvider
-} from '@/modules/workspaces/domain/sso'
+  UserSsoSessionRecord
+} from '@/modules/workspaces/domain/sso/types'
 import Redis from 'ioredis'
 import { Knex } from 'knex'
 import { omit } from 'lodash'
 
 type Crypt = (input: string) => Promise<string>
 
-type StoredSsoProvider = Omit<ProviderRecord, 'provider'> & {
+type SsoProviderRecord = Omit<ProviderRecord, 'provider'> & {
   encryptedProviderData: string
 }
-type WorkspaceSsoProvider = { workspaceId: string; providerId: string }
+type WorkspaceSsoProviderRecord = { workspaceId: string; providerId: string }
 
 const tables = {
-  ssoProviders: (db: Knex) => db<StoredSsoProvider>('sso_providers'),
-  userSsoSessions: (db: Knex) => db<UserSsoSession>('user_sso_sessions'),
+  ssoProviders: (db: Knex) => db<SsoProviderRecord>('sso_providers'),
+  userSsoSessions: (db: Knex) => db<UserSsoSessionRecord>('user_sso_sessions'),
   workspaceSsoProviders: (db: Knex) =>
-    db<WorkspaceSsoProvider>('workspace_sso_providers')
+    db<WorkspaceSsoProviderRecord>('workspace_sso_providers')
 }
 
 export const storeOIDCProviderValidationRequestFactory =
   ({
     redis,
     encrypt
   }: {
-    redis: Redis
+    redis: () => Redis
```

Comment:
Why this needs to be a func?

Reply:
I was doing setup via side-effects on import (not on purpose), so the code was running before Redis was available. Fixed now